### DISCUSSION TITLE
Complete n00b question on RFID Tags

### DISCUSSION THREAD

**Discussion by jamesahendry** [omersiar/esp-rfid#657] [General]
I've got this project up and going on Weigand with a reader and a bag of tags.  Why would every tag come shipped with the same UID - 5200?  I went to the trouble of compiling a board with weigand on Tasmota just to ensure I wasn't going crazy and logs show the same thing.  If I setup a user with 5200 of course I get an unlock/success message.  But something is telling me setting each user up with the same UID probably won't be terribly secure.  Any 101 help appreciated.  They were $4 a bag tag - 10pcs EM4305 T5577 125KHz RFID Rewritable Tag Key Fob Copy Cards.  

https://www.aliexpress.com/item/1005006311913290.html?spm=a2g0o.order_list.order_list_main.22.7d511802dsTvVb

The reader is

https://www.aliexpress.com/item/1005006734543929.html?spm=a2g0o.order_list.order_list_main.47.7d511802dsTvVb
Model: TF9S-EM-W
Card:125KHz EM Card

![image](https://github.com/user-attachments/assets/e040c50a-eb02-4170-9f68-7ee824f55f38)

**Comment by Divyastra** [comment]
@jamesahendry I suspect you are going to need something like this or an equivalent means to write unique UIDs to your key fobs. Being rewritable keyfobs on reflection it sounds reasonable they would all come with the same uid as that would save costs.
https://www.aliexpress.com/item/1005006298903014.html

Alternatively you might want to purchase non-rewritable fobs which should all come with different UIDs
ooo-xxx
